File: VB/WebSite/Bug.vb
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Collections.Generic

Namespace BindToList

    Public Class Bug

        Private idField As Integer

        Public Property Id As Integer
            Get
                Return idField
            End Get

            Set(ByVal value As Integer)
                idField = value
            End Set
        End Property

        Private nameField As String

        Public Property Name As String
            Get
                Return nameField
            End Get

            Set(ByVal value As String)
                nameField = value
            End Set
        End Property

        Private statusField As Integer

        Public Property Status As Integer
            Get
                Return statusField
            End Get

            Set(ByVal value As Integer)
                statusField = value
            End Set
        End Property

        Const count As Integer = 10

        Public Shared Function GetBugList() As List(Of Bug)
            Dim result As List(Of Bug) = New List(Of Bug)(count)
            For i As Integer = 0 To count - 1
                Dim st As Bug = New Bug()
                st.Name = "Bug" & i.ToString()
                st.idField = i
                st.Status = i Mod 3
                result.Add(st)
            Next

            Return result
        End Function
    End Class
End Namespace

File: CS/WebSite/Bug.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Collections.Generic;

namespace BindToList {
    public class Bug {
        private int id;

        public int Id {
            get { return id; }
            set { id = value; }
        }

        private string name;

        public string Name {
            get { return name; }
            set { name = value; }
        }

        private int status;

        public int Status {
            get { return status; }
            set { status = value; }
        }

        const int count = 10;
        public static List<Bug> GetBugList() {
            List<Bug> result = new List<Bug>(count);
            for (int i = 0; i < count; i++) {
                Bug st = new Bug();
                st.Name = "Bug" + i.ToString();
                st.id = i;
                st.Status = i % 3;
                result.Add(st);
            }
            return result;
        }

    }
}
